core/daemon/src/main/java/org/opennms/netmgt/vmmgr/Controller.java

```diff
@@ -179,7 +178,7 @@ static void interpolateSystemProperties() {
                 final String keyStoreKeyProperty = (String) clazz.getField("KEYSTORE_KEY_PROPERTY").get(null);
                 final String defaultKeyStoreKey = (String) clazz.getField("DEFAULT_KEYSTORE_KEY").get(null);
                 secureCredentialsVault = (SecureCredentialsVault) constructor.newInstance(
```

Comment:
Yes, this also need to be adjusted to also consider the SCV type from system properties. Here you also need to read the property files and construct the SCV as needed.